src/main/java/org/duraspace/bagit/BagConfig.java

```diff
@@ -63,31 +63,20 @@
 
     public static final String ACCESS_KEY = "Access";
 
-    private Map<String, Map<String, String>> map;
+    private final Map<String, Map<String, String>> map;
 
     /**
      * Default constructor
      *
      * @param bagConfigFile a bagit config yaml file (see src/test/resources/bagit-config.yml)
+     * @exception IOException if there is an error parsing the bagConfigFile
      */
     @SuppressWarnings("unchecked")
-    public BagConfig(final File bagConfigFile) {
-        final String bagConfigFilePath = bagConfigFile.getAbsolutePath();
-
+    public BagConfig(final File bagConfigFile) throws IOException {
```

Comment:
My general rule is: If the client can recover from an exceptional case, then use a checked exception; otherwise, use unchecked exceptions".

This change seems like a step backwards... however, I recognize that maybe you are working from a different set of assumptions.

Reply:
@awoods For these changes, I was trying to move away from failure as a `RuntimeException` and was certainly looking for feedback wrt how best to inform the client of failure. For the `BagConfig` specifically, having three cases where `RuntimeExceptions` could be thrown felt odd (there was also an assumption that the bag-info needed to be present, which was removed). 

As I'm looking at this again I wonder if a better approach would be to replace the `File bagConfigFile` with `Reader bagConfigReader`, as that pushes any error handling w/ opening the file out of `bagit-support`. This would leave the only exception left from parsing the yaml, which doesn't seem bad.

Reply:
Thanks for the detail, @mikejritter .

Regarding your last point concerning the exception on parsing the yaml. If the client can not recover from a parsing error, I would suggest it be some flavor of RuntimeException.

I am happy to get on the phone if that would be easier.

Reply:
@awoods I believe the latest commits finish resolving this. I updated the last `IOException` to be an `UncheckedIOException` and resolved the conflicts from the previous PR.